### PR TITLE
Remove block buffer and instead use dependencyDag

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

It no longer makes sense to keep the block buffer and
dependencyDag as two separate data structures. Previously,
we needed the block buffer because it stored entire
blocks but now since it only stores hashes, we can extract
the same information from dependencyDag through 

```scala
  private def seenBlockHashes(state: CasperState): Set[BlockHash] =
    state.dependencyDag.parentToChildAdjacencyList.values.toSet.flatten
```

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

https://rchain.atlassian.net/browse/RCHAIN-3147

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by KentShikama** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/util/DoublyLinkedDagOperations.scala [hunk 1]
```diff
@@ -8,6 +8,9 @@ trait DoublyLinkedDag[A] {
   val parentToChildAdjacencyList: Map[A, Set[A]]
   val childToParentAdjacencyList: Map[A, Set[A]]
   val dependencyFree: Set[A]
+  def contains(element: A): Boolean =
```
Comment (rabbitonweb) [inline casper/src/main/scala/coop/rchain/casper/util/DoublyLinkedDagOperations.scala]: this one is new?

**Comment by KentShikama** [comment]
bors try

**Comment by KentShikama** [comment]
Unfortunately, the dependencyDag still needs to keep
information about which blocks actually came in and which
hashes are in the dependencyDag because they are dependencies
of other blocks but aren't yet seen. However, inserting this logic into the DoublyLinkedDag feels off. I'm going to close this for now.

**Pull request closed without merging**